final_prep/mod_zero_hero.rb

```diff
@@ -1,41 +1,57 @@
 # Challenge - See if you can follow the instructions and complete the exercise in under 30 minutes!
 
 # Declare two variables - hero_name AND special_ability - set to strings
-
+hero_name = "dog"
+special_ability = "flying"
 # Declare two variables - greeting AND catchphrase
 #   greeting should be assigned to a string that uses interpolation to include the hero_name
 #   catchphrase should be assigned to a string that uses interpolation to include the special_ability
-
+greeting = "hello i am #{hero_name}!"
+catchphrase = "I can do #{special_ability}"
 # Declare two variables - power AND energy - set to integers
-
+power = 2
+energy = 590
 # Declare two variables - full_power AND full_energy
 #   full_power should multiply your current power by 500
 #   full_energy should add 150 to your current energy
-
+full_power = power * 500
+full_energy = energy * 150
 # Declare two variables - is_human and identity_concealed - assigned to booleans
-
-
+is_human = false
+identity_concealed = true
 # Declare two variables - arch_enemies AND sidekicks
 #   arch_enemies should be an array of at least 3 different enemy strings
 #   sidekicks should be an array of at least 3 different sidekick strings
-
+arch_enemies = ["Mail man", "Vaccum", "Neighbor"]
+sidekicks = ["Chew Toy", "Owner", "Fire Hydrant"]
 # Print the first sidekick to your terminal
-
+p sidekicks.first
 # Print the last arch_enemy to the terminal
-
+p arch_enemies.last
 # Write some code to add a new arch_enemy to the arch_enemies array
-
+arch_enemies << "Veternarian"
 # Print the arch_enemies array to terminal to ensure you added a new arch_enemey
-
+p arch_enemies
 # Remove the first sidekick from the sidekicks array
-
+sidekicks.delete_at(0)
 # Print the sidekicks array to terminal to ensure you added a new sidekick
-
+p sidekicks
 # Create a function called assess_situation that takes three arguments - danger_level, save_the_day, bad_excuse
 #   - danger_level should be an integer
-#   - save_the_day should be a string a hero would say once they save the day 
+#   - save_the_day should be a string a hero would say once they save the day
 #   - bad_excuse should be a string a hero would say if they are too afraid of the danger_level
-
+def assess_situation(danger_level, save_the_day, bad_excuse)
+  danger_level = 9
+  save_the_day = "Ruff that was Ruff"
+  bad_excuse = "Squirel!"
```

Comment:
L44 - 46 are unnecessary here, because `danger_level`, `save_the_day`, and `bad_excuse` are all passed in as arguments.  It would be better to define their values _outside_ of the function declaration, and only pass them in when you **invoke** the function (call the fn).